sdk/core/src/http_client/reqwest.rs

```diff
@@ -0,0 +1,65 @@
+use super::*;
+use std::collections::HashMap;
+
+/// Construct a new `HttpClient` with the `reqwest` backend.
+pub fn new_http_client() -> std::sync::Arc<dyn HttpClient> {
+    std::sync::Arc::new(::reqwest::Client::new())
+}
+
+#[async_trait]
+impl HttpClient for ::reqwest::Client {
+    async fn execute_request(&self, request: &crate::Request) -> crate::Result<crate::Response> {
+        let url = request.url().clone();
+        let mut req = self.request(request.method().clone(), url);
+        for (name, value) in request.headers().iter() {
+            req = req.header(name.as_str(), value.as_str());
+        }
+
+        let body = request.body().clone();
+
+        let reqwest_request = match body {
+            Body::Bytes(bytes) => req
+                .body(bytes)
+                .build()
+                .context(ErrorKind::Other, "failed to build request")?,
+            Body::SeekableStream(mut seekable_stream) => {
+                seekable_stream.reset().await.unwrap(); // TODO: remove unwrap when `HttpError` has been removed
+                req.body(::reqwest::Body::wrap_stream(seekable_stream))
+                    .build()
+                    .context(ErrorKind::Other, "failed to build request")?
+            }
+        };
+
+        let rsp = self
+            .execute(reqwest_request)
+            .await
+            .context(ErrorKind::Io, "failed to execute request")?;
+
+        let status = rsp.status();
+        let headers = to_headers(rsp.headers())?;
+        let body: PinnedStream = Box::pin(rsp.bytes_stream().map_err(|error| {
+            Error::full(
+                ErrorKind::Io,
+                error,
+                "error converting `reqwest` request into a byte stream",
+            )
+        }));
+
+        Ok(crate::Response::new(status, headers, body))
+    }
+}
+
+fn to_headers(map: &::reqwest::header::HeaderMap) -> crate::Result<crate::headers::Headers> {
+    let map = map
+        .iter()
+        .map(|(k, v)| {
+            let key = k.as_str();
+            let value = std::str::from_utf8(v.as_bytes()).expect("non-UTF8 header value");
```

Comment:
I think we probably shouldn't panic here. Maybe for now, we should just skip headers whose values are not UTF8 and log an error.